apps/web/app/api/cancel/route.ts

```diff
@@ -16,8 +17,9 @@ async function handler(req: NextRequest) {
     return NextResponse.json({ success: false, message: "Invalid JSON" }, { status: 400 });
   }
   const session = await getServerSession({ req: buildLegacyRequest(await headers(), await cookies()) });
+  const bookingData = bookingCancelInput.parse(appDirRequestBody);
   const result = await handleCancelBooking({
-    appDirRequestBody,
+    bookingData,
     userId: session?.user?.id || -1,
```

Comment:
Our of scope, but we should return early when there's no logged in user or not care.